spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java

```diff
@@ -71,36 +79,38 @@ public static void main(String[] args) throws Exception {
 			System.err.println("Either the 'modules' system property or 'MODULES' environment variable is required.");
 			System.exit(1);
 		}
-		String moduleHome = System.getProperty("module.home");
-		if (moduleHome == null) {
-			moduleHome = System.getenv("MODULE_HOME");
-		}
-		if (moduleHome == null) {
-			moduleHome = DEFAULT_MODULE_HOME;
-		}
-		ModuleLauncher launcher = new ModuleLauncher(new File(moduleHome));
+		ModuleLauncher launcher = new ModuleLauncher();
 		launcher.launch(modules.split(","), args);
 	}
 
 
 	private static class ModuleLaunchTask implements Runnable {
 
-		private final File directory;
+		private final ModuleResolver moduleResolver;
+
+		private final String groupId;
+
+		private final String artifactId;
 
-		private final String module;
+		private final String version;
 
 		private final String[] args;
 
-		ModuleLaunchTask(File directory, String module, String[] args) {
-			this.directory = directory;
-			this.module = module;
+		ModuleLaunchTask(ModuleResolver moduleResolver, String module, String[] args) {
+			this.moduleResolver = moduleResolver;
+			String[] tokens = module.split(":");
+			Assert.isTrue(tokens.length == 3, "required module format is: 'groupId:artifactId:version'");
```

Comment:
We should avoid sharing that code IMO (for one thing, the ModuleLauncher will ultimately have different ideas for defaults, a bit more opinionated to support OOTB modules as the defaults, with the default group being "org.springframework.cloud.steam" etc). But I did copy the basic parsing code over in the update (2nd commit of this PR).